src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs

```diff
@@ -345,14 +308,7 @@ internal static void RunFromThreadPoolDispatchLoop(Thread threadPoolThread, Exec
             {
                 // The EC always needs to be reset for this overload, as it will flow back to the caller if it performs 
                 // extra work prior to returning to the Dispatch loop. For example for Task-likes it will flow out of await points
-
-                // Restore to Default before Notifications, as the change can be observed in the handler.
-                currentThread.ExecutionContext = null;
-                if (currentExecutionCtx.HasChangeNotifications)
-                {
-                    // There are change notifications; trigger any affected
-                    OnValuesChanged(currentExecutionCtx, nextExecutionCtx: null);
-                }
+                RestoreChangedContextToThread(currentThread, contextToRestore: null, currentExecutionCtx);
```

Comment:
Not related to your changes, but I'm wondering about the `currentExecutionCtx != null` check above.

In what situation will that be null?  Seems like it could be null if the callback suppressed execution context flow, in which case wouldn't we need to restore flow for subsequent work items?

Reply:
As we are inside the ExecutionContext, `null` is the Default context; so its saying if we are no longer on the default (ThreadPool Dispatch starts on Default); then we need to restore

Reply:
Ah, no, wrong "Thread" type.  Was looking at the public one in corefx.  Thanks.